File: src-tauri/src/models/vendor.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Vendor {
    pub id: i64,
    pub tool: String,
    pub name: String,
    pub vendor_key: Option<String>,
    pub base_url: String,
    pub token: String,
    pub model: Option<String>,
    pub config_json: Option<String>,
    pub is_active: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct VendorInput {
    pub name: String,
    pub vendor_key: Option<String>,
    pub base_url: String,
    pub token: String,
    pub model: Option<String>,
    pub config_json: Option<String>,
}

File: src/types/vendor.ts
export interface Vendor {
  id: number;
  tool: string;
  name: string;
  vendor_key: string | null;
  base_url: string;
  token: string;
  model: string | null;
  config_json: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface VendorInput {
  name: string;
  vendor_key: string | null;
  base_url: string;
  token: string;
  model: string | null;
  config_json: string | null;
}

export interface PresetVendor {
  name: string;
  vendor_key: string;
  base_url: string;
  model: string;
}
